apps/consulting/posts/openai-pseudocode.md

```diff
@@ -0,0 +1,225 @@
+specified `review=True` in the decorator which prompted `pseudocode`
+to ask for user feedback on the generated code.
+
+```shell
+What code feedback would you like to provide? (leave empty for approval): this code is messy can you simplify it?
+╭─────────────────────────────────────────────────────────── LLM Feedback ─────────────────────────╮
+│ this code is messy can you simplify it?                                                          │
+╰──────────────────────────────────────────────────────────────────────────────────────────────────╯
+╭──────────────────────────────────────────────────────────── Code Review ─────────────────────────╮
+│                                                                                                  │
+│ import os                                                                                        │
+│ import datetime                                                                                  │
+│ import requests                                                                                  │
+│                                                                                                  │
+│ def run(repository: str) -> list:                                                                │
+│     headers = {'Authorization': f"Bearer {os.environ['GITHUB_TOKEN']}"}                          │
+│     url = f"https://api.github.com/repos/{repository}/issues"                                    │
+│     since = datetime.datetime.now() - datetime.timedelta(days=10)                                │
+│     params = {'since': since.isoformat()}                                                        │
+│     response = requests.get(url, headers=headers, params=params)                                 │
+│     issues = response.json()                                                                     │
+│     return [(i['title'], i['number'], i['created_at']) for i in issues if i['number'] % 2 != 0]  │
+│                                                                                                  │
+╰──────────────────────────────────────────────────────────────────────────────────────────────────╯
+What code feedback would you like to provide? (leave empty for approval):
+```
+
+Already after some quick initial feedback we can see that the LLM
+takes into account the users needs. After user provided feedback the
+automated tests are run. If those tests fail or run into exceptions
+they will be automatically passed back to the LLM for corrections.
+
+```
+pass result = str(all([_[1] % 2 == 1 for _ in run("conda/conda")]))
+github issues [
+  ('Use appropriate `defaults` value for Windows is use when subdir=win-*', 12555, '2023-03-30T17:44:55Z'), 
+  ...
+  ('Document steps to debug tests with vscode', 12459, '2023-03-07T18:20:08Z')
+]
+```
+
+## Conclusion
+
+There are several key ideas here we want to highlight about why we
+think the approaches taken in `pseudocode` are unique.
+
```

Comment:
The first point "Autogenerated code needs guidance" is clear and should be at the start of the bullet.  The second two bullets need similar style in which the bullet starts with a clear point.  "LLMs need declared interfaces"?  "Code declarations and generated code should be separated"?